src/bench/wallet_balance.cpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2012-2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <bench/bench.h>
+#include <interfaces/chain.h>
+#include <key_io.h>
+#include <optional.h>
+#include <test/util.h>
+#include <validationinterface.h>
+#include <wallet/wallet.h>
+
+struct WalletTestingSetup {
+    std::unique_ptr<interfaces::Chain> m_chain = interfaces::MakeChain();
+    CWallet m_wallet;
+
+    WalletTestingSetup()
+        : m_wallet{m_chain.get(), WalletLocation(), WalletDatabase::CreateMock()}
+    {
+    }
+
+    void handleNotifications()
+    {
+        m_wallet.m_chain_notifications_handler = m_chain->handleNotifications(m_wallet);
+    }
+};
+
+static void WalletBalance(benchmark::State& state, const bool set_dirty, const bool add_watchonly, const bool add_mine)
+{
+    const auto& ADDRESS_WATCHONLY = ADDRESS_BCRT1_UNSPENDABLE;
+
+    WalletTestingSetup wallet_t{};
+    auto& wallet = wallet_t.m_wallet;
+    {
+        bool first_run;
```

Comment:
I think there's no need for this to be indented, but I guess it's reasonable to limit the scope the first_run variable.